### PR TITLE
[server][controller][router] Minor cleanups

### PR DESCRIPTION
**Pull request by FelixGV**
PR #1152 added a special behavior to inject latency during the increment operation in AbstractReadResponseStats::incrementMultiChunkLargeValueCount, and this was controlled by a static boolean variable.

The current commit makes this cleaner by removing the special behavior from the main code, and instead using dependency injection in the tests only.

Miscellaneous main code changes:

- Reduced the parameters required to construct a StorageReadRequestHandler.

- The HelixReadOnlySchemaRepository and HelixSchemaAccessor are now less chatty. In particular, listener registration prints one line instead of four.

- Muted the stacktrace for InterruptedExceptions in TopicCleanupService and DictionaryRetrievalService.

- Added a toString to D2Server.

Miscellaneous test changes:

- Added missing @Test annotation to ConsumerIntegrationTestWithProtocolHeader so now it should run as part of the suite.

- Tweaked the integration test bucketing, since bucket Z became the long tail again. Specifially:

  - Moved A/A tests from buckets C to D (where other A/A tests already lived).

  - Placed all consumer-related tests into bucket C (moved from D and Z).

## How was this PR tested?

CI.

## Does this PR introduce any user-facing changes?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: internal/venice-common/src/test/java/com/linkedin/venice/helix/HelixReadOnlySchemaRepositoryTest.java [hunk 1]
```diff
@@ -157,8 +158,8 @@ public void testForceRefreshSchemaData() {
     Assert.assertNull(schemaRepository.getSupersetSchema(storeName));
   }
 
-  @Test
-  public void getDerivedSchemaIdTest() {
+  @Test(dataProviderClass = DataProviderUtils.class, dataProvider = "Two-True-and-False")
```
Comment (sushantmane) [inline internal/venice-common/src/test/java/com/linkedin/venice/helix/HelixReadOnlySchemaRepositoryTest.java:161]: Should we have assertions to check whether the code enabled by `wcEnabled` & `aaEnabled` was called or not?

**Review by sushantmane — COMMENTED** [review]
Thanks, @FelixGV! Overall LGTM, Left just one minor comment

File: internal/venice-common/src/test/java/com/linkedin/venice/helix/HelixReadOnlySchemaRepositoryTest.java (hunk 1 above)
Comment (sushantmane) [reply internal/venice-common/src/test/java/com/linkedin/venice/helix/HelixReadOnlySchemaRepositoryTest.java:162]: Thanks

**Review by sushantmane — APPROVED** [review]
LGTM. Thanks, @FelixGV!

**Comment by FelixGV** [comment]
Thanks @sushantmane!

**Pull request merged**